frontend/app.tsx

```diff
@@ -282,58 +296,41 @@ function diffReducer(state: State, diff: Diff): State {
   if (diff.length === 0) {
     return state;
   }
-  const newAllIssuesMap = new Map(state.allIssuesMap);
-  let newViewIssueCount = state.viewIssueCount;
-  const newFilteredIssues = [...state.filteredIssues];
-  const orderIteratee = partial(getOrderValue, state.issueOrder);
-
-  function add(key: string, newValue: ReadonlyJSONValue) {
-    const newIssue = issueFromKeyAndValue(key, newValue);
-    newAllIssuesMap.set(key, newIssue);
-    if (state.filters.viewFilter(newIssue)) {
-      newViewIssueCount++;
-    }
-    if (state.filters.issuesFilter(newIssue)) {
-      newFilteredIssues.splice(
-        sortedIndexBy(newFilteredIssues, newIssue, orderIteratee),
-        0,
-        newIssue
-      );
-    }
-  }
-  function del(key: string, oldValue: ReadonlyJSONValue) {
-    const oldIssue = issueFromKeyAndValue(key, oldValue);
-    const index = sortedIndexBy(newFilteredIssues, oldIssue, orderIteratee);
-    newAllIssuesMap.delete(key);
-    if (state.filters.viewFilter(oldIssue)) {
-      newViewIssueCount--;
-    }
-    if (newFilteredIssues[index]?.id === oldIssue.id) {
-      newFilteredIssues.splice(index, 1);
-    }
-  }
-  for (const diffOp of diff) {
-    switch (diffOp.op) {
-      case "add": {
-        add(diffOp.key as string, diffOp.newValue);
-        break;
-      }
-      case "del": {
-        del(diffOp.key as string, diffOp.oldValue);
-        break;
-      }
-      case "change": {
-        del(diffOp.key as string, diffOp.oldValue);
-        add(diffOp.key as string, diffOp.newValue);
-        break;
+
+  materialite.tx(() => {
+    for (const diffOp of diff) {
+      switch (diffOp.op) {
+        case "add": {
+          state.issueSource.add(
+            issueFromKeyAndValue(diffOp.key as string, diffOp.newValue)
+          );
+          break;
+        }
+        case "del": {
+          state.issueSource.delete(
+            issueFromKeyAndValue(diffOp.key as string, diffOp.oldValue)
+          );
+          break;
+        }
+        case "change": {
+          state.issueSource.delete(
+            issueFromKeyAndValue(diffOp.key as string, diffOp.oldValue)
+          );
+          state.issueSource.add(
+            issueFromKeyAndValue(diffOp.key as string, diffOp.newValue)
+          );
+          break;
+        }
```

Comment:
@aboodman is this the level of integration you were thinking or something deeper into Replicache?

I'm replacing all the old `diff` logic with calls to the `issueSource`.